src/classes/frModel.cls

```diff
@@ -78,14 +78,16 @@ public with sharing abstract class frModel {
     protected void applyMappings(SObject record, Map<String, Object> request) {
         String sObjectName = record.getSObjectType().getDescribe().getName();
         Map<String, Schema.SObjectField> fields = frSchemaUtil.getFields(sObjectName);
-        
         //a single funraise field might flow into multiple salesforce fields
         Map<String, List<frMapping__c>> frFieldToMappings = new Map<String, List<frMapping__c>>();
         List<frMapping__c> constantMappings = new List<frMapping__c>();
         Set<String> fieldsToQueryFor = new Set<String>();
         Boolean needLastModifiedDate = false;
-        
-        for(frMapping__c mapping : getMappings()) {
+        List<frMapping__c> objMappings = getMappings();
```

Comment:
I don't follow why we call getMappings() here but then re-query them.  Shouldn't getMappings() handle getting the specific type that we need so that we don't need to add branching logic in this base class for each possible type?

Reply:
Fixed! Except gift commitment wouldnt work without it, so leaving that there for now. No functionality should be affected while its there for now while looking for a solution

Reply:
Ok, got it.  I do think that should be resolvable so we'll leave this open until this re-querying can be removed